dlog/plonk/src/verifier.rs

```diff
@@ -0,0 +1,214 @@
+/********************************************************************************************
+
+This source file implements zk-proof batch verifier functionality.
+
+*********************************************************************************************/
+
+pub use super::prover::ProverProof;
+pub use super::index::{VerifierIndex as Index};
+use oracle::{FqSponge, rndoracle::ProofError, utils::PolyUtils, poseidon::SPONGE_BOX, sponge::ScalarChallenge};
+use plonk_circuits::{scalars::{ProofEvaluations, RandomOracles}, constraints::ConstraintSystem};
+use algebra::{Field, PrimeField, AffineCurve, VariableBaseMSM, ProjectiveCurve, Zero, One};
+use commitment_dlog::commitment::{QnrField, CommitmentCurve, PolyComm};
+use ff_fft::{DensePolynomial, EvaluationDomain};
+use crate::plonk_sponge::{FrSponge};
+use rand_core::OsRng;
+
+type Fr<G> = <G as AffineCurve>::ScalarField;
+type Fq<G> = <G as AffineCurve>::BaseField;
+
+impl<G: CommitmentCurve> ProverProof<G> where G::ScalarField : QnrField
+{
+    // This function verifies the batch of zk-proofs
+    //     proofs: vector of Plonk proofs
+    //     index: Index
+    //     RETURN: verification status
+    pub fn verify
```

Comment:
We'll have to do the same thing we're doing in marlin with the commitment to the public input polynomial right?

Reply:
Makes sense. Will do in one of the next commits.

Reply:
This has been completed with commit 3f8916cde2e52bfc15680586a8a153db1c4529e4